### DISCUSSION TITLE
Zephyr port validated boards

### DISCUSSION THREAD

**Discussion by jaenrig-ifx** [micropython/micropython#9098] [Core Development]
Hi there, 

**For the zephyr port, is there anywhere a list validated boards and their supported micropython features (REPL, FS mode, connectivity, gpio, i2c, and hw peripherals, ...)?** 

I already posted this topic with a bit more context on the [micropython forum](https://forum.micropython.org/viewtopic.php?f=3&t=12957), but to increase the chances I try here as well. 

Thanks!

**Comment by jimmo** [comment]
> anywhere a list validated boards

The closest we have right now is the board configs at
https://github.com/micropython/micropython/tree/master/ports/zephyr/boards

> their supported micropython features (REPL, FS mode, connectivity, gpio, i2c, and hw peripherals, ...)?

Unfortunately not that I know of.

My understanding is that any board supported by Zephyr is also support by the Zephyr port, and if that board supports a given feature listed in the [README](https://github.com/micropython/micropython/blob/master/ports/zephyr/README.md) then it will be supported by MicroPython.

i.e. MicroPython basically just forwards everything to Zephyr -- if it has (e.g.) I2C, then MicroPython enables I2C.

**Reply by jaenrig-ifx** [reply]
Thanks @jimmo for your reply :) 

That is my understanding as well. But in practice just calling the micropython application to be compiled for a given board from zephyr end up in different results: from only REPL working, to not building, to only build but no device functioning... 
I have tried the esp32 and rpi pico, both supported in zephyr, and natively in micropython. 
I did not thoroughly check, but maybe not all the resources are really enabled on zephyr side.
Thus, either a little bit more of experience (mine is low) is required on zephyr side to pass the right building flags, overlays, config files... or for most cases some additional integration work is required. 

The only boards that we can trust to be really compiling as micropython is being further developed are the ones under CI:
https://github.com/micropython/micropython/blob/13dceaa4ea4336eddb763985c2274cfa497c8e82/tools/ci.sh#L698-L703

After some research in forums, issues, etc. it seems the frdm_k64f board has been mostly used to developed the zephyr. So that would be a good reference to try out. Unfortunately it is out of stock in Europe :(

**Reply by MaureenHelm** [reply]
In theory any board supported by Zephyr should work, but in practice some boards may need a Kconfig overlay to configure features (e.g., `ports/zephyr/boards/frdm_k64f.conf`). The `frdm_k64f` board would be a good board to start with; I've also used MicroPython with `mimxrt1050_evk` and `reel_board` in the past.

If you're still having trouble with your board of choice, you can try building with the minimal configuration (`prj_minimal.conf`) and incrementally add features from the normal configuration (`prj.conf`) back in. It's possible the networking configuration may not work on your board.
```
$ west build -b <your board name> -- -DCONF_FILE=prj_minimal.conf`
```

Check which version of Zephyr you're using with MicroPython. I haven't tested MicroPython with the latest main Zephyr branch lately, so I recommend using the last Zephyr release (v3.1.0)

**Reply by jaenrig-ifx** [reply]
Hi @MaureenHelm,

Thanks a lot for your support 😃 I managed to get the REPL, and the machine module (pin and i2c so far) working with the nrf52840dk board. While that I found (what I believe) a major compatibility issue [#9115](https://github.com/micropython/micropython/issues/9115) for new versions of zephyr in which I already mentioned you. Your opinion would be very valuable there.
